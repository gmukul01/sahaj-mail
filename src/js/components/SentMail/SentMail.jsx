import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import EmailList from 'components/email/EmailList';

const Inbox = props => {
	const { selectedEmails, addToSelectedEmails } = selectedEmailsState([]);
	const { emails, fetchEmails } = props;

	useEffect(() => {
		fetchEmails();
	}, []);

	function onRead() {}

	return (
		<section className="content sentMail">
			<EmailList isSentMail {...{ selectedEmails, addToSelectedEmails, onRead, emails }} />
		</section>
	);
};

Inbox.propTypes = {
	totalEmails: PropTypes.number,
	emails: PropTypes.array,
	fetchEmails: PropTypes.func,
	deleteEmails: PropTypes.func,
	readEmails: PropTypes.func,
	fetchInboxDetails: PropTypes.func
};

export default React.memo(Inbox);

const selectedEmailsState = initialValue => {
	const [emails, setValue] = useState(initialValue);

	function handleChange(e) {
		const emailId = Number(e.target.value),
			newSelectedEmails = emails.indexOf(emailId) === -1 ? [...emails, emailId] : emails.filter(id => id !== emailId);

		setValue(newSelectedEmails);
	}

	return { selectedEmails: emails, addToSelectedEmails: handleChange };
};
