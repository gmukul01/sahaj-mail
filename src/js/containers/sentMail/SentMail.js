import { connect } from 'react-redux';

import SentMail from 'components/SentMail/SentMail';
import { fetchEmails } from 'actions/emails';

const mapStateToProps = ({ sent: { emails, pageNumber, emailsPerPage } }) => ({
	emails,
	pageNumber,
	emailsPerPage
});

const mergeProps = (stateProps, dispatchProps, ownProps) => {
	const { fetchEmails } = dispatchProps;
	const { pageNumber, emailsPerPage } = stateProps;
	return {
		...stateProps,
		...dispatchProps,
		...ownProps,
		fetchEmails() {
			fetchEmails('sent', pageNumber, emailsPerPage);
		}
	};
};

export default connect(
	mapStateToProps,
	{ fetchEmails },
	mergeProps
)(SentMail);
