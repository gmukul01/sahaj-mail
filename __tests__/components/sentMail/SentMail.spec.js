import React from 'react';
import renderer from 'react-test-renderer';

import SentMail from 'components/sentMail/SentMail';

describe('SentMail Component', () => {
	const mockFetchEmail = jest.fn(),
		initialProps = {
			totalUnread: 2,
			emails: [],
			fetchEmails: mockFetchEmail
		};

	afterEach(() => {
		mockFetchEmail.mockClear();
	});

	it('should match snapshot', () => {
		const inbox = renderer.create(<SentMail {...initialProps} />).toJSON();
		expect(inbox).toMatchSnapshot();
	});
});
