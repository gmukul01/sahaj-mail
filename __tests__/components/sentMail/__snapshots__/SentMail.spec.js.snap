// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentMail Component should match snapshot 1`] = `
<section
  className="content sentMail"
>
  <div
    className="email-list"
  />
</section>
`;
